diff --git a/bin/CMakeLists.txt b/bin/CMakeLists.txt
index cc0b4fa15a..9d8955c464 100644
--- a/bin/CMakeLists.txt
+++ b/bin/CMakeLists.txt
@@ -1,11 +1,35 @@
 get_property(triton_libs GLOBAL PROPERTY TRITON_LIBS)
 
+include_directories(
+    ${PROJECT_SOURCE_DIR}/third_party/ascend/AscendNPU-IR/bishengir/include
+    ${PROJECT_BINARY_DIR}/third_party/ascend/AscendNPU-IR/bishengir/include
+  )
+
 add_llvm_executable(triton-opt triton-opt.cpp PARTIAL_SOURCES_INTENDED)
 
 # TODO: what's this?
 llvm_update_compile_flags(triton-opt)
 target_link_libraries(triton-opt PRIVATE
   ${triton_libs}
+  # Ascend backend libraries
+  DiscreteMaskAccessConversion
+  TritonToAnnotation
+  TritonToHFusion
+  TritonToHIVM
+  TritonToLinalg
+  TritonToLLVM
+  TritonToStructured
+  TritonToUnstructure
+  # BiShengIR dialects
+  BiShengIRAnnotationDialect
+  BiShengIRHACCDialect
+  BiShengIRHFusionDialect
+  BiShengIRHIVMDialect
+  BiShengIRMathExtDialect
+  BiShengIRMemRefExtDialect
+  BiShengIRScopeDialect
+  BiShengIRSymbolDialect
+  BiShengIRTensorDialect
   # tests
   TritonTestAnalysis
   TritonTestDialect
@@ -27,6 +51,25 @@ mlir_check_all_link_libraries(triton-reduce)
 llvm_update_compile_flags(triton-reduce)
 target_link_libraries(triton-reduce PRIVATE
   ${triton_libs}
+  # Ascend backend libraries
+  DiscreteMaskAccessConversion
+  TritonToAnnotation
+  TritonToHFusion
+  TritonToHIVM
+  TritonToLinalg
+  TritonToLLVM
+  TritonToStructured
+  TritonToUnstructure
+  # BiShengIR dialects
+  BiShengIRAnnotationDialect
+  BiShengIRHACCDialect
+  BiShengIRHFusionDialect
+  BiShengIRHIVMDialect
+  BiShengIRMathExtDialect
+  BiShengIRMemRefExtDialect
+  BiShengIRScopeDialect
+  BiShengIRSymbolDialect
+  BiShengIRTensorDialect
   # tests
   TritonTestAnalysis
   TritonTestDialect
@@ -47,6 +90,25 @@ add_llvm_executable(triton-lsp triton-lsp.cpp PARTIAL_SOURCES_INTENDED)
 llvm_update_compile_flags(triton-lsp)
 target_link_libraries(triton-lsp PRIVATE
   ${triton_libs}
+  # Ascend backend libraries
+  DiscreteMaskAccessConversion
+  TritonToAnnotation
+  TritonToHFusion
+  TritonToHIVM
+  TritonToLinalg
+  TritonToLLVM
+  TritonToStructured
+  TritonToUnstructure
+  # BiShengIR dialects
+  BiShengIRAnnotationDialect
+  BiShengIRHACCDialect
+  BiShengIRHFusionDialect
+  BiShengIRHIVMDialect
+  BiShengIRMathExtDialect
+  BiShengIRMemRefExtDialect
+  BiShengIRScopeDialect
+  BiShengIRSymbolDialect
+  BiShengIRTensorDialect
   # tests
   TritonTestAnalysis
   TritonTestDialect
@@ -86,6 +148,26 @@ export_executable_symbols_for_plugins(triton-llvm-opt)
 add_llvm_executable(triton-tensor-layout triton-tensor-layout.cpp PARTIAL_SOURCES_INTENDED)
 target_link_libraries(triton-tensor-layout PRIVATE
   ${triton_libs}
+  # Ascend backend libraries
+  DiscreteMaskAccessConversion
+  TritonToAnnotation
+  TritonToHFusion
+  TritonToHIVM
+  TritonToLinalg
+  TritonToLLVM
+  TritonToStructured
+  TritonToUnstructure
+  # BiShengIR dialects
+  BiShengIRAnnotationDialect
+  BiShengIRHACCDialect
+  BiShengIRHFusionDialect
+  BiShengIRHIVMDialect
+  BiShengIRMathExtDialect
+  BiShengIRMemRefExtDialect
+  BiShengIRScopeDialect
+  BiShengIRSymbolDialect
+  BiShengIRTensorDialect
+  # tests
   TritonTestAnalysis
   TritonTestDialect
   TritonTestProton
diff --git a/bin/RegisterTritonDialects.h b/bin/RegisterTritonDialects.h
index 48049f0268..8f8fe8ea46 100644
--- a/bin/RegisterTritonDialects.h
+++ b/bin/RegisterTritonDialects.h
@@ -1,6 +1,36 @@
 #pragma once
+#include "ascend/include/AutoBlockify/Passes.h"
+#include "ascend/include/DiscreteMaskAccessConversion/Passes.h"
+#include "ascend/include/DynamicCVPipeline/AddControlFlowCondition.h"
+#include "ascend/include/DynamicCVPipeline/AllocMultiCache/AddMultiBufferInnerScope.h"
+#include "ascend/include/DynamicCVPipeline/AllocMultiCache/AddMultiBufferOuterScope.h"
+#include "ascend/include/DynamicCVPipeline/AnalyzeDataFlow.h"
+#include "ascend/include/DynamicCVPipeline/ComputeBlockOptPass.h"
+#include "ascend/include/DynamicCVPipeline/Passes.h"
+#include "ascend/include/DynamicCVPipeline/RemoveAttributes.h"
+#include "ascend/include/DynamicCVPipeline/SeparateMemoryFromCompute/AddMultiBufferToGMLoadPass.h"
+#include "ascend/include/DynamicCVPipeline/SeparateMemoryFromCompute/AsyncLoadHoistingPass.h"
+#include "ascend/include/DynamicCVPipeline/SplitDataflow/RefineArgsBlockId.h"
+#include "ascend/include/DynamicCVPipeline/StandardizeOp.h"
+#include "ascend/include/TritonControlFlowOpt/Passes.h"
+#include "ascend/include/TritonToAnnotation/Passes.h"
+#include "ascend/include/TritonToGraph/Passes.h"
+#include "ascend/include/TritonToHFusion/Passes.h"
+#include "ascend/include/TritonToHIVM/Passes.h"
+#include "ascend/include/TritonToLLVM/Passes.h"
+#include "ascend/include/TritonToLinalg/Passes.h"
+#include "ascend/include/TritonToStructured/Passes.h"
+#include "ascend/include/TritonToUnstructure/Passes.h"
+
+#include "bishengir/Dialect/Annotation/IR/Annotation.h"
+#include "bishengir/Dialect/HACC/IR/HACC.h"
+#include "bishengir/Dialect/HFusion/IR/HFusion.h"
+#include "bishengir/Dialect/HIVM/IR/HIVM.h"
+#include "bishengir/Dialect/Scope/IR/Scope.h"
+
 #include "amd/include/Dialect/TritonAMDGPU/IR/Dialect.h"
 #include "amd/include/TritonAMDGPUTransforms/Passes.h"
+#include "ascend/include/Dialect/TritonAscend/IR/TritonAscendDialect.h"
 #include "nvidia/include/Dialect/NVGPU/IR/Dialect.h"
 #include "nvidia/include/Dialect/NVWS/IR/Dialect.h"
 #include "proton/Dialect/include/Conversion/ProtonGPUToLLVM/Passes.h"
@@ -85,6 +115,19 @@ inline void registerTritonDialects(mlir::DialectRegistry &registry) {
   mlir::triton::registerConvertTritonGPUToLLVMPass();
   mlir::triton::registerConvertNVGPUToLLVMPass();
   mlir::triton::registerAllocateSharedMemoryNvPass();
+  mlir::triton::registerTritonToLinalgPasses();
+  mlir::triton::registerTritonControlFlowOptPasses();
+  mlir::triton::registerDiscreteMaskAccessConversion();
+  mlir::triton::registerTritonToStructuredPasses();
+  mlir::triton::registerTritonToAnnotationPasses();
+  mlir::triton::registerTritonToUnstructurePasses();
+  mlir::triton::registerTritonToHIVMPasses();
+  mlir::triton::registerTritonToHFusionPasses();
+  mlir::triton::registerTritonToLLVMPasses();
+  mlir::triton::registerAutoBlockifyPasses();
+  mlir::triton::cfg::registerTritonToGraphPasses();
+  mlir::triton::registerBubbleUpOperationPass();
+
   mlir::registerLLVMDIScope();
   mlir::LLVM::registerInlinerInterface(registry);
   mlir::NVVM::registerInlinerInterface(registry);
@@ -136,16 +179,39 @@ inline void registerTritonDialects(mlir::DialectRegistry &registry) {
   mlir::triton::proton::gpu::registerScheduleBufferStorePass();
   mlir::triton::proton::gpu::registerAddSchedBarriersPass();
 
+  // DynamicCVPipeline passes
+  mlir::triton::registerAddDynamicCVPipelinePasses();
+  mlir::triton::registerPreCheckAvailablePasses();
+  mlir::triton::registerStandardizeOpPasses();
+  mlir::triton::registerAddControlFlowConditionPasses();
+  mlir::triton::registerAddMultiBufferOuterScopePasses();
+  mlir::triton::registerAddMultiBufferInnerScopePasses();
+  mlir::triton::registerRemoveSsbufAttrPasses();
+  mlir::triton::registerAnalyzeDataFlowPasses();
+  mlir::triton::registerComputeBlockOptPasses();
+  mlir::triton::registerPlanComputeBlockPasses();
+  mlir::triton::registerOpClassifierPass();
+  mlir::triton::registerRefineArgsBlockIdPasses();
+  mlir::triton::registerAsyncLoadHoistingPasses();
+  mlir::triton::registerAddMultiBufferToGMLoadPasses();
+
   registry.insert<
       mlir::triton::TritonDialect, mlir::cf::ControlFlowDialect,
       mlir::triton::nvidia_gpu::TritonNvidiaGPUDialect,
       mlir::triton::gpu::TritonGPUDialect,
       mlir::triton::instrument::TritonInstrumentDialect,
       mlir::math::MathDialect, mlir::arith::ArithDialect, mlir::scf::SCFDialect,
-      mlir::gpu::GPUDialect, mlir::LLVM::LLVMDialect, mlir::NVVM::NVVMDialect,
+      mlir::tensor::TensorDialect, mlir::gpu::GPUDialect,
+      mlir::LLVM::LLVMDialect, mlir::NVVM::NVVMDialect,
       mlir::triton::nvgpu::NVGPUDialect, mlir::triton::nvws::NVWSDialect,
       mlir::triton::amdgpu::TritonAMDGPUDialect,
       mlir::triton::proton::ProtonDialect,
       mlir::triton::proton::gpu::ProtonGPUDialect, mlir::ROCDL::ROCDLDialect,
-      mlir::triton::gluon::GluonDialect>();
+      mlir::triton::gluon::GluonDialect,
+      mlir::triton::ascend::TritonAscendDialect, mlir::hivm::HIVMDialect,
+      mlir::scope::ScopeDialect, mlir::hacc::HACCDialect,
+      mlir::annotation::AnnotationDialect, mlir::hfusion::HFusionDialect,
+      mlir::tensor::TensorDialect, mlir::linalg::LinalgDialect,
+      mlir::memref::MemRefDialect, mlir::bufferization::BufferizationDialect,
+      mlir::func::FuncDialect>();
 }
diff --git a/bin/triton-opt.cpp b/bin/triton-opt.cpp
index 2d2570771a..91ed8b79ae 100644
--- a/bin/triton-opt.cpp
+++ b/bin/triton-opt.cpp
@@ -1,4 +1,5 @@
 #include "./RegisterTritonDialects.h"
+#include "ascend/include/DynamicCVPipeline/AllocMultiCache/AddMultiBufferInnerScope.h"
 
 #include "mlir/Tools/mlir-opt/MlirOptMain.h"
 
@@ -6,6 +7,9 @@ int main(int argc, char **argv) {
   mlir::DialectRegistry registry;
   registerTritonDialects(registry);
 
+  // Register AddMultiBufferInnerScope pass
+  mlir::triton::registerAddMultiBufferInnerScopePasses();
+
   return mlir::asMainReturnCode(mlir::MlirOptMain(
       argc, argv, "Triton (GPU) optimizer driver\n", registry));
 }
diff --git a/python/triton/runtime/autotuner.py b/python/triton/runtime/autotuner.py
index 0c4d710496..d98afb86f9 100644
--- a/python/triton/runtime/autotuner.py
+++ b/python/triton/runtime/autotuner.py
@@ -126,7 +126,7 @@ class Autotuner(KernelInterface):
         return self._do_bench
 
     def _bench(self, *args, config, **meta):
-        from ..compiler.errors import CompileTimeAssertionFailure
+        from ..compiler.errors import CompileTimeAssertionFailure, MLIRCompilationError
 
         verbose = knobs.autotuning.print
         if verbose:
@@ -162,7 +162,7 @@ class Autotuner(KernelInterface):
 
         try:
             return self.do_bench(kernel_call, quantiles=(0.5, 0.2, 0.8))
-        except (OutOfResources, CompileTimeAssertionFailure, PTXASError) as e:
+        except (OutOfResources, CompileTimeAssertionFailure, MLIRCompilationError) as e:
             if verbose:
                 print(f"Autotuning failed with {e}")
             return [float("inf"), float("inf"), float("inf")]
